armi/reactor/converters/uniformMesh.py

```diff
@@ -269,32 +286,254 @@ def makeAssemWithUniformMesh(sourceAssem, newMesh):
+                    f"be reported to the developers."
+                )
+
+            # Iterate over each of the blocks that were found in the uniform mesh
+            # source assembly within the lower and upper bounds of the destination
+            # block and perform the parameter mapping.
+            updatedDestVals = collections.defaultdict(float)
+
+            _setNumberDensitiesFromOverlaps(destBlock, sourceBlocksInfo)
+
+            for setter, getter, paramNames in [
+                (
+                    BlockParamMapper.scalarParamSetter,
+                    BlockParamMapper.scalarParamGetter,
+                    blockScalarParamNames,
+                ),
+                (
+                    BlockParamMapper.arrayParamSetter,
+                    BlockParamMapper.arrayParamGetter,
+                    blockArrayParamNames,
+                ),
+            ]:
+
+                for sourceBlock, sourceBlockOverlapHeight in sourceBlocksInfo:
+                    sourceBlockVals = getter(sourceBlock, paramNames)
+                    sourceBlockHeight = sourceBlock.getHeight()
+
+                    for paramName, sourceBlockVal in zip(paramNames, sourceBlockVals):
+                        # The value can be `None` if it has not been set yet. In this case,
+                        # the mapping should be skipped.
+                        if sourceBlockVal is None:
+                            continue
+
+                        # Determine if the parameter is volumed integrated or not.
+                        isVolIntegrated = sourceBlock.p.paramDefs[paramName].atLocation(
+                            parameters.ParamLocation.VOLUME_INTEGRATED
+                        )
+
+                        # If the parameter is volume integrated (e.g., flux, linear power)
+                        # then calculate the fractional contribution from the source block.
+                        if isVolIntegrated:
+                            integrationFactor = (
+                                sourceBlockOverlapHeight / sourceBlockHeight
+                            )
+
+                        # If the parameter is not volume integrated (e.g., volumetric reaction rate)
+                        # then calculate the fraction contribution on the destination block.
+                        # This smears the parameter over the destination block.
+                        else:
+                            integrationFactor = (
+                                sourceBlockOverlapHeight / destinationBlockHeight
+                            )
+
+                        updatedDestVals[paramName] += sourceBlockVal * integrationFactor
+
+                    setter(destBlock, updatedDestVals.values(), updatedDestVals.keys())
+
+                # For parameters that have not been set on the destination block, recover these
+                # back to their originals based on the cached values.
+                for paramName in paramNames:
```

Comment:
suggest making this a private method called
_applyCachedValues 
or something like that